source/slang/parameter-binding.cpp

```diff
@@ -2432,13 +2432,13 @@ static void collectEntryPointParameters(
     context->shared->programLayout->entryPoints.Add(entryPointLayout);
 
     // Note: this isn't really the best place for this logic to sit,
-    // but it is the simplest place where we have a direct correspondance
+    // but it is the simplest place where we have a direct correspondence
     // between a single `EntryPointRequest` and its matching `EntryPointLayout`,
     // so we'll use it.
     //
     for( auto taggedUnionType : entryPoint->taggedUnionTypes )
     {
-        auto substType = taggedUnionType->Substitute(typeSubst).As<Type>();
+        auto substType = taggedUnionType->Substitute(typeSubst).dynamicCast<Type>();
```

Comment:
There are various cases like this that should probably eventually be a `checkedCast` or similar, that performs the `dynamicCast` and then `assert`s that the result is non-null.